netboot/main.go

```diff
@@ -165,15 +172,50 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	}
 	debug("DHCP: boot file URL is %s", bootfile)
 	// check for supported schemes
-	if !strings.HasPrefix(bootfile, "http://") {
-		return fmt.Errorf("DHCP: can only handle http scheme")
+	matched, err := regexp.Match("^(http|https)://", []byte(bootfile))
```

Comment:
nit: can you make these two blocks a separate function which returns the httpclient? this boot function is getting white long already.

Reply:
done!